lib/arch/avx2/dec_loop.c

```diff
@@ -20,26 +20,44 @@ while (srclen >= 45)
 	//  5  [97..122]  [26..51]  -71  a..z
 	// (6) Everything else => invalid input
 
-	const __m256i set1 = CMPEQ(str, '+');
-	const __m256i set2 = CMPEQ(str, '/');
-	const __m256i set3 = RANGE(str, '0', '9');
-	const __m256i set4 = RANGE(str, 'A', 'Z');
-	const __m256i set5 = RANGE(str, 'a', 'z');
-
-	__m256i delta = REPLACE(set1, 19);
-	delta = _mm256_or_si256(delta, REPLACE(set2,  16));
-	delta = _mm256_or_si256(delta, REPLACE(set3,   4));
-	delta = _mm256_or_si256(delta, REPLACE(set4, -65));
-	delta = _mm256_or_si256(delta, REPLACE(set5, -71));
-
-	// Check for invalid input: if any of the delta values are zero,
-	// fall back on bytewise code to do error checking and reporting:
-	if (_mm256_movemask_epi8(CMPEQ(delta, 0))) {
+	// code by @aqrit from
+	// https://github.com/WojciechMula/base64simd/issues/3#issuecomment-271137490
+	// transated into AVX2
```

Comment:
The code in AVX2 is the same than in SSSE3. Fully explained in SSSE3.
@aqrit shared his code on WojciechMula/base64simd and lemire/fastbase64 thus it's BSD licensed. He did not share his full name to get proper copyright notice though. He's mentioned in the commit and PR to be notified.